### ISSUE TITLE
I'm trying to build TensorFlow Servings using bazel 0.3.0 , keeps halting the procedure.

### ISSUE THREAD

**Issue by mch5048** [tensorflow/serving#126]
To begin with, sorry for my insufficient context and content for the troubleshooting, since I am a newbie 

for programming and not accustomed to using English......

My system is Ubuntu 16.04 and python 2.7.12? .

I followed the official installation instruction from the site, and 

I'm on the verge of building the TF Servings with the bazel as the installation guide says.

However, whenever I type

bazel build tensorflow_serving/...

the log shows really really numerous warnings like 
................................................................................................................................................
external/org_tensorflow/tensorflow/core/kernels/tensor_array_ops.cc:615:74:   required from 'void tensorflow::TensorArrayConcatOp<Device, T>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::ThreadPoolDevice; T = unsigned char]'
external/org_tensorflow/tensorflow/core/kernels/tensor_array_ops.cc:1060:1:   required from here
external/org_tensorflow/tensorflow/core/kernels/tensor_array.h:222:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
external/org_tensorflow/tensorflow/core/kernels/tensor_array.h: In instantiation of 'tensorflow::Status tensorflow::TensorArray::ReadMany(tensorflow::OpKernelContext*, std::vector<tensorflow::PersistentTensor>*, tensorflow::int32) [with Device = Eigen::ThreadPoolDevice; T = short int; tensorflow::int32 = int]':
external/org_tensorflow/tensorflow/core/kernels/tensor_array_ops.cc:615:74:   required from 'void tensorflow::TensorArrayConcatOp<Device, T>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::ThreadPoolDevice; T = short int]'
external/org_tensorflow/tensorflow/core/kernels/tensor_array_ops.cc:1060:1:   required from here
external/org_tensorflow/tensorflow/core/kernels/tensor_array.h:222:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
external/org_tensorflow/tensorflow/core/kernels/tensor_array.h: In instantiation of 'tensorflow::Status tensorflow::TensorArray::ReadMany(tensorflow::OpKernelContext*, std::vector<tensorflow::PersistentTensor>*, tensorflow::int32) [with Device = Eigen::ThreadPoolDevice; T = short unsigned int; tensorflow::int32 = int]':
external/org_tensorflow/tensorflow/core/kernels/tensor_array_ops.cc:615:74:   required from 'void tensorflow::TensorArrayConcatOp<Device, T>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::ThreadPoolDevice; T = short unsigned int]'
external/org_tensorflow/tensorflow/core/kernels/tensor_array_ops.cc:1060:1:   required from here
external/org_tensorflow/tensorflow/core/kernels/tensor_array.h:222:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
.........................................................................................................................................
these.....

I googled and tried to fix it for several hours but stil stucked here.

What could be the problem? Please give me some helps!

**Comment by kirilg** [comment]
These are compile time warnings in core TensorFlow (which we include as a submodule). Your title says "keep halting the procedure", did you get an actual error further down or did compilation succeed, albeit with a lot of warnings? Note that if you rerun the `bazel build` command again right after, it won't print the warnings the second time unless compilation fails.

**Comment by mch5048** [comment]
@kirilg Thanks for  your reply! My sytem kept halting so I needed to reboot always. I get it around with the --local_resources_256... . It seems to the build succeed but, Im still suspicious that I can do healthy further programming. Is there any way I can manage these kind of problem?

**Comment by schaaf23** [comment]
Probably your system does not hang, it is just incredibly slow?
did you try to cancel the build with ctrl-c? it will take some time, but you will see if your system is still alive when bazel responds to the cancellation.
I had a similar issue when compiling TensorFlow, lowering the number of CPUs with 

`--local_resources 3000,0.5,1 `

helped.  
You have to adjust these numbers to fit your ressources. But if you like to actually continue working on the machine as you compile, setting the number of CPUs for bazel (second parameter) to less than half of the number of CPUs in your system helps. My VM has two cores, and with a value of 0.5 I still had a max load of more than 10 while compiling. With a value of 1 it turned so slow at some stage, that I had to reboot it (although it was still alive, but i had to wait 10 minutes to enter a single char at the console)
Basically bazel should be able to take care of the ressources, but for small systems bazels choice seems a bit "optimistic".

also consider using the `-c opt` flag, as TensorFlow will end up beeing very slow without

**Comment by kirilg** [comment]
Thanks @schaaf23. That's right, if --local_resources worked, that likely means that the issue is with machine resources and capping the amount you allow Bazel to use should unblock you (at the cost of likely increasing total compilation time since resources are capped). Other than setting that flag, I don't think there is any way around it short of getting a more powerful machine.
